Update developer guide (#678)
jupyter-book/myst-theme: docs/developer/tips.md

# Developer tips and best practices

## Using Providers for MyST content

> When using components that render MyST content (using `<MyST ast={content} />`), they must be wrapped wrap them in various providers e.g. `TabStateProvider` to ensure proper React context is available.

Reviewer: I think this has little to do with `MyST` components. It might be something like:

`Providers` create [React contexts](https://react.dev/reference/react/createContext) for communicating app states across components. For example,
the `<ArticleProvider>...</ArticleProvider>` shares an `ArticleContext` with all its children, telling them the current article `kind`, `frontmatter`, and `references`.

Reviewer: This one still, then merge?

> These providers facilitate an alternative to prop drilling, the name used to describe passing props down through several React components. They are also used to isolate state between different subtrees, e.g. `TabStateProvider` defines a synchronised tab context.

Reviewer: ```suggestion
These providers facilitate an alternative to prop drilling, the name used to describe passing props down several layers of React components.
```

>  They are also used to isolate state between different subtrees, e.g. `TabStateProvider` defines a synchronised tab context.

Maybe that should stay, but I'm not sure user has ability to understand without further explanation.
